#! /bin/sh -e

# DP: hotshot/pstats.py: Error out on missing profile/pstats modules.
# DP: Add a note to the library documentation, that the profile and pstats
# DP: modules can be found in the python2.x-profiler package.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Doc/library/profile.rst~	2008-04-18 20:39:55.000000000 +0200
+++ Doc/library/profile.rst	2008-05-30 11:07:19.000000000 +0200
@@ -5,6 +5,12 @@
 The Python Profilers
 ********************
 
+Debian note: The license for the :mod:`profile` and :mod:`pstats`
+modules doesn't conform to the Debian Free Software Guidelines (DFSG).
+These modules can be found in the *python-profiler* package in the
+*non-free* section of the Debian archives or in the the *multiverse*
+section of the Ubuntu archives.
+
 .. sectionauthor:: James Roskind
 
 
@@ -241,6 +247,12 @@
    :synopsis: Python profiler
 
 
+Debian note: The license for the :mod:`profile` and :mod:`pstats`
+modules doesn't conform to the Debian Free Software Guidelines (DFSG).
+These modules can be found in the *python-profiler* package in the
+*non-free* section of the Debian archives or in the the *multiverse*
+section of the Ubuntu archives.
+
 The primary entry point for the profiler is the global function
 :func:`profile.run` (resp. :func:`cProfile.run`). It is typically used to create
 any profile information.  The reports are formatted and printed using methods of
